test/s3path.jl

```diff
@@ -421,6 +421,42 @@ function s3path_tests(config)
         end
     end
 
+    @testset "isdir" begin
+        # Top level
+        path = S3Path("s3://$(bucket_name)"; config=config)
+        @test isdir(path) == true
+
+        # No Such Bucket
+        test_exception = AWSException(
+            "NoSuchBucket",
+            "",
+            nothing,
+            AWS.HTTP.Exceptions.StatusError(404, "", "", ""),
+            nothing,
+        )
+        patch = @patch AWSS3.S3.list_objects_v2(args...; kwargs...) = throw(test_exception)
+
+        path = S3Path("s3://some_non_existent_bucket_7051649213"; config=config)
+        apply(patch) do
+            @test isdir(path) == false
+        end
+
+        # Other Error
+        test_exception = AWSException(
+            "TestException",
+            "",
+            nothing,
+            AWS.HTTP.Exceptions.StatusError(404, "", "", ""),
+            nothing,
+        )
+        patch = @patch AWSS3.S3.list_objects_v2(args...; kwargs...) = throw(test_exception)
+
+        path = S3Path("s3://$(bucket_name)"; config=config)
+        apply(patch) do
+            @test_throws AWSException isdir(path)
+        end
+    end
+
```

Comment:
```suggestion
    @testset "isdir" begin
        function _generate_exception(code)
            return AWSException(
                code,
                "",
                nothing,
                AWS.HTTP.Exceptions.StatusError(404, "", "", ""),
                nothing,
            )
        end

        const s3_path = S3Path("s3://$(bucket_name"; config=config)

        @testset "top level" begin
            @test isdir(s3_path) == true
        end
        
        @testset "no such bucket" begin
            test_exception = _generate_exception("NoSuchBucket")
            patch = @patch AWSS3.S3.list_objects_v2(args...; kwargs...) = throw(test_exception)

            apply(patch) do
                @test isdir(s3_path) == false
            end
        end

        @testset "other error thrown" begin
            test_exception = _generate_exception("TestException")
            patch = @patch AWSS3.S3.list_objects_v2(args...; kwargs...) = throw(test_exception)
            apply(patch) do
                @test_throws AWSException isdir(s3_path)
            end
        end
    end

```

Reply:
Modified it a bit since all the tests are for top level bucket access.